### ISSUE TITLE
useless-super-delegation when typing differs

### ISSUE THREAD

**Issue by z33ky** [PyCQA/pylint#1923]
### Steps to reproduce

```
class Egg():
    def __init__(self, thing: object) -> None: ...

class Spam(Egg):
    def __init__(self, thing: int) -> None:
        super().__init__(thing)
```

This is useful so type-checkers can check that `Spam` is only ever created with `int`, while `Egg`s can be created with anything.

### Current behavior

A warning is emitted: `Useless super delegation in method '__init__' (useless-super-delegation)`.

### Expected behavior

No warning.

### pylint --version output

```
No config file found, using default configuration
pylint 1.8.2,
astroid 1.6.1
Python 3.6.4 (default, Jan  5 2018, 02:35:40)
[GCC 7.2.1 20171224]
```

**Comment by brycepg** [comment]
I think it would be reasonable to only raise this warning when the annotations are the same

**Comment by AWhetter** [comment]
To any new contributors looking to tackle this issue; [the classes checker](https://github.com/PyCQA/pylint/blob/1c0356f0683901e6e3561f4caeb176ff3e20be1e/pylint/checkers/classes.py#L865) raises the `useless-super-delegation` message. `_definition_equivalent_to_call` will need to start considering the annotations of the [arguments node](http://astroid.readthedocs.io/en/latest/api/astroid.nodes.html#astroid.nodes.Arguments) of the function.
